stage1/enter/enter.c

```diff
@@ -50,17 +51,48 @@ static int openpidfd(int pid, char *which) {
 int main(int argc, char *argv[])
 {
 	int	fd;
-	int	pid;
+	int	pid = 0;
+	char *appname = NULL;
 	pid_t	child;
 	int	status;
 	int	root_fd;
 
+	int c;
+
 	/* The parameters list is specified in
 	 * Documentation/devel/stage1-implementors-guide.md */
-	exit_if(argc < 4,
-		"Usage: %s pid appName cmd [args...]", argv[0])
+	while (1) {
+		int option_index = 0;
+		static struct option long_options[] = {
+		   {"pid",     required_argument, 0,  'p' },
+		   {"appname", required_argument, 0,  'a' },
+		   {0,         0,                 0,  0 }
+		};
+
+		c = getopt_long(argc, argv, "p:a:",
+				long_options, &option_index);
+		if (c == -1)
+			break;
+
+		switch (c) {
+		case 'p':
+			pid = atoi(optarg);
+			break;
+		case 'a':
+			appname = optarg;
+			break;
+		case 0:
+			break;
+		case ':':   /* missing option argument */
+		case '?':
+		default:
+			fprintf(stderr, "Usage: %s --pid=1234 "
+					"--appname=name -- cmd [args...]",
+					argv[0]);
+			exit(1);
+		}
```

Comment:
FWIW I intentionally left this minimal and unfriendly.  It's an internal stage1 implementation detail, not an interactive utility and I didn't want to imply otherwise by having this kind of stuff present.